txtorcon/circuit.py

```diff
@@ -275,3 +275,25 @@ def __str__(self):
         path = ' '.join([x.ip for x in self.path])
         return "<Circuit %d %s [%s] for %s>" % (self.id, self.state, path,
                                                 self.purpose)
+
+class CircuitBuildTimedOutError(Exception):
+    """
+    This exception is thrown when using `timed_circuit_build`
+    and the circuit build times-out.
+    """
+
+def build_timeout_circuit(tor_state, clock, path, timeout):
+    """
```

Comment:
"in general" I try to pass in reactor references and never do "from twisted.internet import reactor".

So, the factory methods like ``build_tor_connection`` get a reactor passed in, but so far there's no reason for TorState or TorControlProtocol to save a reactor ref. So I think the answer is "no, it doesn't already have a ref".